addons/website_slides/static/src/js/slides_course_page.js

```diff
@@ -0,0 +1,148 @@
+/** @odoo-module **/
+
+import publicWidget from 'web.public.widget';
+import session from 'web.session';
+import { qweb as QWeb } from 'web.core';
+
+
+/**
+ * Global widget for both fullscreen view and non-fullscreen view of a slide course.
+ * Contains general methods to update the UI elements (progress bar, sidebar...) as well
+ * as method to mark the slide as completed / uncompleted.
+ */
+export const SlideCoursePage = publicWidget.Widget.extend({
+    events: {
+        'click button.o_wslides_button_complete': '_onClickComplete',
+    },
+
+    custom_events: {
+        'slide_completed': '_onSlideCompleted',
+        'slide_to_complete': '_onSlideToComplete',
+    },
+
+    xmlDependencies: ['/website_slides/static/src/xml/website_slides_sidebar.xml'],
+
+    /**
+     * Greens up the bullet when the slide is completed
+     *
+     * @public
+     * @param {Integer} slideId
+     * @param {Boolean} completed
+     */
+    setSlideCompleted: function (slide, completed = true) {
+        const $button = this.$(`.o_wslides_sidebar_done_button[data-id=${slide.id}]`);
+
+        const newButton = QWeb.render('website.slides.sidebar.done.button', {
```

Comment:
This implies having 2 separate templates (JS + pytho,) and re-render the whole thing every time we toggle.

Wild idea: would it be complicated to instead only render in python and then attach a Widget to the DOM that handles the toggle?

That way we only have once the template and we move this logic to a small contained Widget.